packages/devtools_app/lib/src/inspector/layout_explorer/box/box.dart

```diff
@@ -0,0 +1,428 @@
+    final length = sizes.length;
+    double total = 0.0;
+    total += 1.0; // Add a tiny amount to avoid divide by zero bugs.
+    final fractions = minFractions.toList();
+    for (var size in sizes) {
+      if (size != null) {
+        total += math.max(0, size);
+      }
+    }
+
+    double totalFraction = 0.0;
+    for (int i = 0; i < length; i++) {
+      final size = sizes[i];
+      if (size != null) {
+        fractions[i] = math.max(size / total, minFractions[i]);
+        totalFraction += fractions[i];
+      } else {
+        fractions[i] = 0.0;
+      }
+    }
+    if (totalFraction != 1.0) {
+      for (int i = 0; i < length; i++) {
+        fractions[i] = fractions[i] / totalFraction;
+      }
+    }
+    final output = <double>[];
+    for (var fraction in fractions) {
+      output.add(fraction * availableSize);
+    }
+    return output;
+  }
+
+  Widget _buildChild(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+    final parentProperties = this.parentProperties ??
+        properties; // Fall back to this node's properties if there is no parent.
+
+    final parentSize = parentProperties.size;
+    final offset = properties.node?.parentData ??
+        (BoxParentData()..offset = const Offset(0, 0));
+
+    if (properties.size == null) {
+      // This should happen infrequently but it is better to show an error than
+      // crash.
+      return Center(
+        child: Text(
+          'Visualizing layouts for ${properties.description} widgets is not yet supported.',
+        ),
+      );
+    }
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        // Subtract out one pixel border on each side.
+        final availableHeight = constraints.maxHeight - 2;
+        final availableWidth = constraints.maxWidth - 2;
+
+        final minFractions = [0.2, 0.3, 0.2];
+        double nullOutZero(double value) => value != 0.0 ? value : null;
+        final widths = [
+          nullOutZero(offset.offset.dx),
+          properties.size.width,
+          nullOutZero(parentSize != null
+              ? parentSize.width - (properties.size.width + offset.offset.dx)
+              : 0.0),
+        ];
+        final heights = [
+          nullOutZero(offset.offset.dy),
+          properties.size.height,
+          nullOutZero(parentSize != null
+              ? parentSize.height - (properties.size.height + offset.offset.dy)
+              : 0.0),
+        ];
+        final displayWidths = minFractionLayout(
+          availableSize: availableWidth,
+          sizes: widths,
+          minFractions: minFractions,
+        );
+        final displayHeights = minFractionLayout(
+          availableSize: availableHeight,
+          sizes: heights,
+          minFractions: minFractions,
+        );
+        final widgetWidth = displayWidths[1];
+        final widgetHeight = displayHeights[1];
```

Comment:
why index 1?

Reply:
commented on what the 3 elements in the widths and heights array are.